core/views.py

```diff
@@ -341,3 +345,82 @@ class SourceDetail(generics.RetrieveUpdateDestroyAPIView):
 
 class LoginView(KnoxLoginView):
     authentication_classes = [BasicAuthentication]
+
+
+class ScheduledEventList(APIView):
+    """
+    List all Scheduled Events, or create a new Scheduled Event.
+    """
+    def get(self, request, format=None):
+        scheduled_events = ScheduledEvent.objects.all()
+        serializer = ScheduledEventSerializer(scheduled_events, many=True)
+        return Response(serializer.data)
+
+    def post(self, request, format=None):
+        serializer = ScheduledEventSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ScheduledEventRunInstanceList(APIView):
+    """
+    List all ScheduledEventRunInstances, or create a new ScheduledEventRunInstance.
+    """
+    def get(self, request, format=None):
+        scheduled_event_run_instance = ScheduledEventRunInstance.objects.all()
+        serializer = ScheduledEventRunInstanceSerializer(scheduled_event_run_instance, many=True)
+        return Response(serializer.data)
+
+    def post(self, request, format=None):
+        serializer = ScheduledEventRunInstanceSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
This requires a scheduled event as context, otherwise we'll just be fetching all instances, and creating instances that aren't attached to an event, which aren't useful.

Reply:
Now using `pk` path variable from `scheduled_event/<int:pk>/run_instances` which is an event id to attach events to run instances